deepchem/models/torch_models/infograph.py

```diff
@@ -101,81 +291,138 @@ class InfoGraphModel(ModularTorchModel):
         self.average_loss = average_loss
+        self.training_mode = training_mode
+        if training_mode == 'supervised':
+            self.embedding_dim = embedding_dim
+            self.use_unsup_loss = False
+            self.separate_encoder = False
+        elif training_mode == 'semisupervised':
+            self.embedding_dim = embedding_dim * num_gc_layers
+            self.use_unsup_loss = True
+            self.num_gc_layers = num_gc_layers
+            self.separate_encoder = True
 
         self.components = self.build_components()
         self.model = self.build_model()
         super().__init__(self.model, self.components, **kwargs)
 
     def build_components(self):
-        return {
-            'encoder':
-                InfoGraphEncoder(self.num_features, self.edge_features,
-                                 self.embedding_dim),
-            'unsup_encoder':
-                InfoGraphEncoder(self.num_features, self.edge_features,
-                                 self.embedding_dim),
-            'ff1':
-                MultilayerPerceptron(2 * self.embedding_dim, self.embedding_dim,
-                                     (self.embedding_dim,)),
-            'ff2':
-                MultilayerPerceptron(2 * self.embedding_dim, self.embedding_dim,
-                                     (self.embedding_dim,)),
-            'fc1':
-                torch.nn.Linear(2 * self.embedding_dim, self.embedding_dim),
-            'fc2':
-                torch.nn.Linear(self.embedding_dim, 1),
-            'local_d':
-                MultilayerPerceptron(self.embedding_dim,
-                                     self.embedding_dim, (self.embedding_dim,),
-                                     skip_connection=True),
-            'global_d':
-                MultilayerPerceptron(2 * self.embedding_dim,
-                                     self.embedding_dim, (self.embedding_dim,),
-                                     skip_connection=True),
-        }
+        """
+        Builds the components of the InfoGraphStar model. InfoGraphStar works by maximizing the mutual information between the graph-level representation and the representations of substructures of different scales.
+
+        It does this by producing graph-level encodings and substructure encodings, and then using a discriminator to classify if they are from the same molecule or not.
+
+        The encoder is a graph convolutional network that produces the graph-level encodings and substructure encodings.
+
+        In a supervised training mode, only 1 encoder is used and the encodings are not compared, while in a semi-supvervised training mode they are different in order to prevent negative transfer from the pretraining stage.
+
+        The local discriminator is a multilayer perceptron that classifies if the substructure encodings are from the same molecule or not while the global discriminator classifies if the graph-level encodings are from the same molecule or not.
```

Comment:
My apologies for the confusion, but I meant that the explicit names of the components should be documented in the API. The description here is useful too though! We should have both